### PR TITLE
add jsesc options support

### LINKED ISSUE #8124: Feature Request: jsesc option for babel-generator
## Feature Request

**Is your feature request related to a problem? Please describe.**
I am using babel-generator to generate some codes for ASTs. But I want to customize the `jsesc` option. Currently the option is mostly fixed:

```javascript
// @babel/generator
// src/generators/types.js

// ensure the output is ASCII-safe
const opts = {
  quotes: "double",
  wrap: true,
};
if (this.format.jsonCompatibleStrings) {
   opts.json = true;
}
const val = jsesc(node.value, opts);
```
I want to specific the `opts` variable in the global option of `generate` function. **I will submit a PR if you think it's OK.**

**Describe the solution you'd like**
The usage of `generate` function:
```javascript
import {parse} from '@babel/parser';
import generate from '@babel/generator';

const code = 'class Example {}';
const ast = parse(code);

const output = generate(ast, {
  jsescOption: {/* option */},
}, code);
```
And the exist jsesc related option `jsonCompatibleStrings` would be reserved. And if the jsescOption is `null` or `undefined`, it will use default option. And the option object will also be merged to default option.

**Describe alternatives you've considered**

None.

**Teachability, Documentation, Adoption, Migration Strategy**

None.

### PR DESCRIPTION
**Pull request by vincentdchan**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #8124  
| Patch: Bug Fix?          | No
| Major: Breaking Change?  | No
| Minor: New Feature?      | Yes
| Tests Added + Pass?      | No
| Documentation PR         | No
| Any Dependency Changes?  | No
| License                  | MIT

### CODE REVIEW

File: packages/babel-generator/src/index.js [hunk 1]
```diff
@@ -53,6 +53,11 @@ function normalizeOptions(code, opts): Format {
       base: 0,
     },
     decoratorsBeforeExport: !!opts.decoratorsBeforeExport,
+    jsescOption: {
+      quotes: "double",
+      wrap: true,
+      ...(opts.jsescOption || {}),
```
Comment (nicolo-ribaudo) [inline packages/babel-generator/src/index.js]: `|| {}` isn't needed; object spread works with `null`/`undefined`/`false`

Comment (vincentdchan) [reply packages/babel-generator/src/index.js]: fixed

**Comment by hzoo** [comment]
I realize that with this change we can just remove the `jsonCompatibleStrings` option? 

Instead users can do `{ jsescOption: { json: true } }`?

**Comment by vincentdchan** [comment]
@hzoo Of cause it can, but it would be a breaking change, I don't know if babel can accept it?

**Pull request merged**